### ISSUE TITLE
Can we run Liri OS on RPi 3?

### ISSUE THREAD

**Issue by yangxuan8282** [lirios/lirios#24]
I know this maybe kind of heavy for pi, but just want to ask the possibility.
If it's possible, will wayland and VC 4 improve the performance?

**Comment by plfiorini** [comment]
Yes I'm pretty sure it's possible to run it on RPi 3, I did on the RPi 2 back when it was Hawaii.
I don't think VC 4 will yield better performance compared to brcm.
I'm building an image for RPi 2 with VC 4. First I need to verify if VC 4 can work to begin with then I will need to check how it runs.

**Comment by yangxuan8282** [comment]
thanks for reply
waiting for your image release 
I have another question that maybe off the topic, but it really trouble me for months 
since the liri os have QT 5 dependency, can you share compiling QT 5 ( 5.7 or 5.8, with qtwayland and qtwebkit ) recipe on raspberry pi 3 ( I guess it should be same for RPi3 and RPi2 )  
have tried several times, but all failed
and all instruction on google is for older version

**Comment by plfiorini** [comment]
Build recipes should work on both RPi3 and RPi2. For embedded images I'm experimenting with Yocto (see https://github.com/lirios/meta-liri) that has common recipes for every target and adds machine specific bits on top. To be even more specific I have used the same recipes to build a test image for Intel before diving into RPi 2. It's pretty neat since general purpose distros are usually limited in that regard, for instance GL drivers are often a bit painful as those distros tend to rely on Mesa so as soon as you need a proprietary GL driver you have to overwrite the libraries.

**Comment by plfiorini** [comment]
Managed to get a Liri OS build on Raspberry Pi 2 with VC4.
It's been a bit painful, had to upgrade mesa to make QtQuick programs work.
Performance is way better than I remember back when it was Hawaii on Arch with the propretary driver, boots as fast as 20/25s but VC4 is not reliable yet.

As soon as I managed to polish the image I will decide if it's worth to upload a whopping 1,5GB file to our serve and release it as technology preview.

**Comment by yangxuan8282** [comment]
great, I used to saw your hawaii image links for RPi2, but no chance to try it, because the links is expired

**Comment by yangxuan8282** [comment]
Do you think if it's possible to distribute the Liri OS desktop environment with docker? KDE Neon have do something similar : https://community.kde.org/Neon/Docker

**Comment by plfiorini** [comment]
Yes it's doable, already started working on it :D But there are a few problems with d-bus to sort out.
![screenshot from 2017-02-23 22-58-24](https://cloud.githubusercontent.com/assets/843710/23280972/a380d828-fa1b-11e6-967f-80b2393513c7.png)

**Comment by yangxuan8282** [comment]
Will you release the RPi image soon?

**Comment by R3d-Sky** [comment]
Unfortunately, I tried building it on master except lirios-wayland on my pi3, but it does not work. It fails to Initialize EGL display on both vc4 and brcm.

**Comment by plfiorini** [comment]
The RPi 2 image is here: https://repo.liri.io/images/raspberry-pi/

It's unofficial, very rough just an experiment. VC4 is not very stable yet.

**Comment by yangxuan8282** [comment]
@plfiorini finally, thanks

**Comment by plfiorini** [comment]
I don't want to release it officially because: VC4 is unreliable but at the same time I don't feel like investing time for brcm (proprietary driver) anymore given that a FLOSS alternative being in development.
So keep that in mind.

**Comment by yangxuan8282** [comment]
sorry for late reply, my lapdock is broken and I just got new monitor, so just got chance to test it today

login the image with root, and start liri with:

```
liri-session
```

I can see the desktop, and could launch setting, but failed to launch liri terminal

also have tried to install liri DE on archlinux arm with aur package [liri-shell-git](https://aur.archlinux.org/packages/liri-shell-git/), but can't start liri, it complaint something like:  Could not find DRM device, have no idea how to fix it

thanks for your works, and wait for 1.0 coming

**Comment by wizofe** [comment]
@plfiorini How do you think this image would work on a Raspi 3? Could I offer any help around?

**Comment by plfiorini** [comment]
@wizofe Can you burn the image on a SDCard and see if it boots?

**Comment by wizofe** [comment]
Where can I find the image though? The link has no image in.

**Comment by plfiorini** [comment]
@wizofe It was deleted to make some room I didn't even remeber. Luckily I had a copy on my hard drive and I have uploaded it again in the same directory.

**Comment by wizofe** [comment]
@plfiorini thanks for the re-upload. Just tried with RasPi3 but it just shows a rainbow boot screen and then goes...black. I am really keen on porting it to RasPi3, I am aware that they 've been few changes in the architecture and I am willing to check through the Datasheet of the processor. 

I have not big experience in OS programming though, so I would be happy if you guide me a bit on what files I need to change etc.

**Comment by Leon0402** [comment]
I tried it on my raspberry 3b too and can confirm wizofe`s experience, just a black screen.
Would be really cool if you are able to fix it.

By the way, I tried liri os some weeks ago on my computer and it`s really awsome, but for some reasons I couldn´t open the settings :( 
I am not quite sure if the problem still exists, but I can try it if you have a new image. I am really looking forward to the stable version!

Best regards, Leon
